### PR TITLE
Add beamforming modules

### PR DESCRIPTION
**Pull request by mpariente**
This is inspired by the code in [`nn-gev`](https://github.com/fgnt/nn-gev/blob/f8287020b18a00dd34eb58ba439b8d7dfad74fac/fgnt/beamforming.py) from Paderborn and François Grondin's code in [speechbrain](https://github.com/speechbrain/speechbrain/blob/develop/speechbrain/processing/decomposition.py).
Thanks for open-sourcing it. 
Also helped by @popcornell for MWF

### Left to code: 
- GCCPhat with quadratic interpolation (cf François Grondin's code)
- steering vector 
- Blind analytic normalization 
- GEV vector phase correction

### In parallel 
- More docs
- Examples
- Tests

Might be interesting for @boeddeker @FrancoisGrondin @faroit

### CODE REVIEW

**Comment by popcornell** [comment]
GCCPhat with quadratic interpolation (cf François Grondin's code). 

We might want to do a Localization module for that.

**Comment by mpariente** [comment]
Makes sense for the localisation, you're right. 

By the way, eternal question, should we make a functional version, and module wrappers? 
It might be easier to use for functional addicts

**Comment by popcornell** [comment]
Can you be more specific ? 
Module wrappers you mean wrapping an STFT encoder module in the case of GCC-PHAT ?

**Comment by mpariente** [comment]
Sorry. 

Currently, the beamforming code it's only accessible in nn.Modules. 
In PyTorch, all ops are first implemented as functions, and then it also provides some class wrappers. 

Should we outsource everything in functions, and only make the function calls in the class, so bothb usage are possible?

**Comment by mpariente** [comment]
@popcornell could you review please?
